### ISSUE TITLE
Separate selfaware ?

### ISSUE THREAD

**Issue by George3d6** [mindsdb/lightwood#217]
The gist of this pondering boils down to these 2 things:

a) Currently we can't start training the selfaware when we start training the model, otherwise the model never converges unless our lr is unreasonably slow (think 1e-6 kinda values)

b) Currently the loss from the selfaware seems to often be much greater than that of the actual model and the whole setup makes training less stable (gradients go all over the place in different epochs instead of constantly going down)

c) Currently the selfaware network is part of the `DefaultNet` making that whole setup much more complicated... but I see no reason why it ought to be

d) Currently when the selfaware kicks in we have to lower the main networks learning rate, thus making it slower.

There seems to be an obvious solution to all of this problem:

Make the selfaware a separate network, with it's own optimizer.

This means that we can start training it right away and it won't hamper training the main model in the beginning. This means the selfaware can be trained for longer

It also means that we can just adjust the selfaware's optimizer (or weight down it's loss, though that seems like a much worst idea) to a lower learning rate and keep the main networks optimizer to the highest lr possible.

The code basically becomes something like:

```
Yh = net(X)
loss = criterion(Yh,Y)
loss.backwards()
opt.step()

awareness_in = torch.cat(Yh,X)
if not train_together:
   awareness_in = awareness_in.detach()
awareness = selfaware(awareness_in)
# Remember, in the current setup the selfaware is trying to predict the unreduced loss of the main model's predictions
unreduced_loss = criterion(Yh,Y)
selfaware_loss = selfaware_criterion(awareness, unreduced_loss)
selfaware_loss.backwards()
selfaware_opt.step()
```

Suddenly enabling the selfware and disabling it also becomes much simple, no need to reinitialize the whole main network, just a single if is needed to stop backpropagatin it's results through the main model and another one to stop it from making predictions and/or being trained.

**Comment by paxcema** [comment]
Implemented in #228.
